sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -188,24 +197,80 @@ private AnalyzeSentimentResult convertToAnalyzeSentimentResult(DocumentSentiment
+     * @param includeOpinionMining The boolean indicator to include opinion mining data in the returned result. If this
+     * flag is specified, you'll get a {@code minedOpinions} property on SentenceSentiment. It's available start from
+     * v3.1-preview.1 service version.
      * @param options The {@link TextAnalyticsRequestOptions} request options.
      * @param context Additional context that is passed through the Http pipeline during the service call.
      *
      * @return A mono {@link Response} contains {@link AnalyzeSentimentResultCollection}.
      */
     private Mono<Response<AnalyzeSentimentResultCollection>> getAnalyzedSentimentResponse(
-        Iterable<TextDocumentInput> documents, TextAnalyticsRequestOptions options, Context context) {
-        // TODO: add opinion mining in the following PR
+        Iterable<TextDocumentInput> documents, boolean includeOpinionMining, TextAnalyticsRequestOptions options,
+        Context context) {
         return service.sentimentWithResponseAsync(
             new MultiLanguageBatchInput().setDocuments(toMultiLanguageInput(documents)),
             options == null ? null : options.getModelVersion(),
             options == null ? null : options.isIncludeStatistics(),
-            null,
+            includeOpinionMining,
             context.addData(AZ_TRACING_NAMESPACE_KEY, COGNITIVE_TRACING_NAMESPACE_VALUE))
             .doOnSubscribe(ignoredValue -> logger.info("A batch of documents - {}", documents.toString()))
             .doOnSuccess(response -> logger.info("Analyzed sentiment for a batch of documents - {}", response))
             .doOnError(error -> logger.warning("Failed to analyze sentiment - {}", error))
             .map(this::toAnalyzeSentimentResultCollectionResponse)
             .onErrorMap(throwable -> mapToHttpResponseExceptionIfExist(throwable));
     }
+
+    /*
+     * Transform SentenceSentiment's opinion mining to output that user can use.
+     */
+    private IterableStream<MinedOpinion> toMinedOpinionList(
+        com.azure.ai.textanalytics.implementation.models.SentenceSentiment sentenceSentiment) {
+        // If include opinion mining indicator is false, the service return null for the aspect list.
+        if (sentenceSentiment.getAspects() == null) {
+            return null;
+        }
+        final List<SentenceAspect> sentenceAspects = sentenceSentiment.getAspects();
+        final List<MinedOpinion> minedOpinions = new ArrayList<>();
+        sentenceAspects.forEach(sentenceAspect -> {
+            final List<OpinionSentiment> opinionSentiments = new ArrayList<>();
+            sentenceAspect.getRelations().forEach(aspectRelation -> {
+                final AspectRelationType aspectRelationType = aspectRelation.getRelationType();
+                final String refLink = aspectRelation.getRef();
+                final int refIndex = Integer.parseInt(refLink.substring(refLink.lastIndexOf("/") + 1));
```

Comment:
what happens if the the character "/" is not found?

Reply:
will reconsider the checking. How to parse the reflink will need to revisited.

Reply:
added the support to parse full reference pointer. See findSentimentOpinion()